stew/result.nim

```diff
@@ -151,12 +151,14 @@ type
       v: T
 
 func raiseResultError[T, E](self: Result[T, E]) {.noreturn.} =
+  mixin toException
+
   when E is ref Exception:
     if self.e.isNil: # for example Result.default()!
       raise ResultError[void](msg: "Trying to access value with err (nil)")
     raise self.e
-  elif compiles(self.e.toException()):
-    raise self.e.toException()
+  elif compiles(toException(self.e)):
```

Comment:
This change is probably not strictly necessary, but I've learned to mistrust the use of the dot operator in generic procs.